pdns/slavecommunicator.cc

```diff
@@ -618,6 +618,33 @@ void CommunicatorClass::suck(const DNSName &domain, const ComboAddress& remote,
       notifyDomain(domain, &B);
     }
 
+    // Execute LUA-AXFR-END-SCRIPT script
+    unique_ptr<AuthLua4> laes{nullptr};
+    vector<string> axfr_end_scripts;
+    string axfr_end_script=::arg()["lua-axfr-end-script"];
+    if(B.getDomainMetadata(domain, "LUA-AXFR-END-SCRIPT", axfr_end_scripts) && !afxr_end_scripts.empty()) {
+      if (pdns_iequals(afxr_end_scripts[0], "NONE")) {
+        afxr_end_script.clear();
+      } else {
+        afxr_end_script=afxr_end_scripts[0];
+      }
+    }
+    if(!axfr_end_script.empty()){
+      try {
+        laes = make_unique<AuthLua4>();
+        laes->loadFile(axfr_end_script);
+        g_log<<Logger::Info<<logPrefix<<"loaded Lua script '"<<afxr_end_script<<"'"<<endl;
+      }
+      catch(std::exception& e) {
+        g_log<<Logger::Error<<logPrefix<<"failed to load Lua script '"<<axfr_end_script<<"': "<<e.what()<<endl;
+        return;
+      }
+    }
+    
+    if (!laes) {
+      laes->axfr_finished(domain);
+      g_log<<Logger::Info << logPrefix << "Executed LUA-AXFR-END-SCRIPT for zone: " << domain << endl;
```

Comment:
might be a bit too verbose on busy servers

Reply:
https://github.com/m0rcq/pdns/blob/feature/axfr/lua-axfr-end/pdns/slavecommunicator.cc#L639 set to debug to reduce noise except explicitly required.